core/src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala

```diff
@@ -213,28 +232,104 @@ case class UpdateCommand(
       spark, txn, "update", rootPath, inputLeafFiles, nameToAddFileMap)
     val newTarget = DeltaTableUtils.replaceFileIndex(target, baseRelation.location)
     val targetDf = Dataset.ofRows(spark, newTarget)
-    val updatedDataFrame = {
-      val updatedColumns = buildUpdatedColumns(condition)
-      targetDf.select(updatedColumns: _*)
-    }
+
+    val updatedDataFrame = withUpdatedColumns(
+      target,
+      updateExpressions,
+      condition,
+      targetDf.withColumn(CONDITION_COLUMN_NAME, new Column(condition)),
```

Comment:
We are adding the new column here but dropping the column in `withUpdatedColumns`. Is it cleaner to just pass the targetDf and let the `withUpdatedColumns` build on the `targetDf`?